libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -148,6 +230,11 @@ namespace eosio { namespace chain {
 
       const chainbase::database& db;
       const wasm_interface::vm_type wasm_runtime_time;
+      mutable std::shared_mutex instantiation_cache_mutex;
```

Comment:
`mutable` members always stick out a bit to me. It seems like the only reason this is `mutable` is because of `is_code_cached() const`. But maybe this is a good use of `mutable` --  `is_code_cached()` clearly wants to indicate its "presented" (externally accessible) state isn't modified by `is_code_cached()`, so it does seem to make some sense to "exclude" mutexes from `const`ness. Just curious to hear a little more what others think about this pattern... tbh I didn't look around and see if it's well established elsewhere in our code.

Reply:
I do think mutexes is the perfect example of a valid use of mutable.